HPCC-24018 Add IncludeServiceNames option into WUInfo

wangkx opened the pull request:
Default to false.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

wangkx commented:
@afishbeck @GordonSmith please review.

HPCCSmoketest commented:
Process of PR-13955, label: h24018 starts now.
The reason of this test is: Forced to re-test.
Commit ID: b241c0bcfd540c1c924792ffc0f86a7d4ec579c5
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-219.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-219.ca-central-1.compute.internal
Sha: b241c0bcfd540c1c924792ffc0f86a7d4ec579c5
Build: success
Milestone:Install hpccsystems-platform-community_7.10.3-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |75 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |43 sec  (00:00:43) |
|setup (roxie) |11 |11 |0 |18 sec  (00:00:18) |
|test (hthor) |883 |883 |0 |308 sec  (00:05:08) |
|test (thor) |821 |821 |0 |762 sec  (00:12:42) |
|test (roxie) |962 |962 |0 |406 sec  (00:06:46) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  749 sec (00:12:29) |  120 sec (00:02:00) |   22 sec (00:00:22) |   20 sec (00:00:20) | 1826 sec (00:30:26) |   18 sec (00:00:18) | 2764 sec (00:46:04) |

afishbeck commented on esp/services/ws_workunits/ws_workunitsHelpers.cpp:
```diff
@@ -898,6 +898,38 @@ void WsWuInfo::getGraphTimingData(IArrayOf<IConstECLTimingData> &timingData)
+    ForEach(*graphs)
+    {
+        Owned <IPropertyTree> xgmml = graphs->query().getXGMMLTree(false);
+        Owned<IPropertyTreeIterator> iter = xgmml->getElements("//node[@label='SOAP dataset']");
```
@wangkx 
Technically the label is a description and could change, the real identifier is the ` <att name="_kind" value="55"/>` 
Where value is a few different possibilities.
But I realize that filtering by an att named value may be very slow.
If you can't filter by kind, the label should be stable at least for now.

Currently the label can be several values which also complicates your filtering.
"SOAP dataset", "SOAP action", "HTTP dataset"

wangkx replied:
Could we assume that the WUInfo user knows the label and add the label to WUInfoRequest?

jakesmith replied:
I would have thought it should be by activity kind ..
But then should this service be agnostic about what it's been asked to include, i.e. why just a particular set of kinds?
Should the service cope with a list of activity types to find? So e.g. could request CSV Write activities instead.

I'm a bit concerned about the efficiency and requirements here.
The result will be a flattened (i.e. you will no longer be able to tell which graph/subgraph they came from) list of activities found, with all their attributes.
Ultimately how is the info. consumed by the client requesting it?
@GordonSmith @wangkx

afishbeck replied:
@jakesmith it's unfortunate that kind is an attribute of node.  Fits in with a discussion of how inefficient xgmml can be.
Is there an efficient way of filtering nodes that fall into a set of activity kinds?  SOAP and HTTP call alone have a total of 5 kinds.
I don't think this use case is about understanding the graph of the query.  It's more about knowing what services etc the query calls out to.  If the list of soap and http calls were cached somewhere more accessible that would work to.

jakesmith commented:
>It's more about knowing what services etc the query calls out to. If the list of soap and http calls were cached somewhere more accessible that would work too.

ok that makes sense. I guess SOAP type activities are a good proxy for knowing what services are related to a query (PIPE is another possible route).

>Is there an efficient way of filtering nodes that fall into a set of activity kinds? SOAP and HTTP call alone have a total of 5 kinds.

There isn't an efficient IPT/xpath way. Multiple IPT/xpath calls to get each one would be very inefficient.
But //node is going to end up pulling everything (and getGraphs has already done so anyway).
Walking the nodes and filtering by a set of _kind's wouldn't add much overhead.
i.e. something like:
```
Owned<IPropertyTreeIterator> iter = xgmml->getElements("//node");
ForEach(*iter)
{
    unsigned kind = iter.query().getPropInt("att[@name=\"_kind\"]/@value);
    if (kind)
    {
        const char *kindStr = getActivityText(kind);
        if (seachSet.find(kindStr) != searchSet.end())
        {
             // add
        }
    }
}


```

jakesmith commented:
The above would be a bit more efficient if the search set contained the kind's as unsigned's.

jakesmith commented:
>It's more about knowing what services etc the query calls out to.
>SOAP and HTTP call alone have a total of 5 kinds.

@afishbeck - should this request members/methods be called getServiceActivities or similar vs "SOAPDataset"* ?

afishbeck commented:
Yes that name would make more sense to me.

HPCCSmoketest commented:
Process of PR-13955, label: h24018 starts now.
The reason of this test is: Did not build yet.
Commit ID: b241c0bcfd540c1c924792ffc0f86a7d4ec579c5
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: b241c0bcfd540c1c924792ffc0f86a7d4ec579c5
Build: success
Milestone:Install hpccsystems-platform-community_7.10.3-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |76 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |30 sec  (00:00:30) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |883 |883 |0 |584 sec  (00:09:44) |
|test (thor) |821 |821 |0 |1025 sec  (00:17:05) |
|test (roxie) |962 |962 |0 |730 sec  (00:12:10) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   11 sec (00:00:11) | 1056 sec (00:17:36) |  124 sec (00:02:04) |   23 sec (00:00:23) |   17 sec (00:00:17) | 2718 sec (00:45:18) |   17 sec (00:00:17) | 3966 sec (01:06:06) |

HPCCSmoketest commented:
Process of PR-13955, label: h24018 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 52e79e1993cf142726b23ded0c87a1bbe2a65227
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-168.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-168.ca-central-1.compute.internal
Sha: 52e79e1993cf142726b23ded0c87a1bbe2a65227
Build: success
Milestone:Install hpccsystems-platform-community_7.10.3-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |72 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |26 sec  (00:00:26) |
|setup (thor) |11 |11 |0 |42 sec  (00:00:42) |
|setup (roxie) |11 |11 |0 |18 sec  (00:00:18) |
|test (hthor) |883 |883 |0 |310 sec  (00:05:10) |
|test (thor) |821 |821 |0 |748 sec  (00:12:28) |
|test (roxie) |962 |962 |0 |396 sec  (00:06:36) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  742 sec (00:12:22) |  118 sec (00:01:58) |   23 sec (00:00:23) |   18 sec (00:00:18) | 1799 sec (00:29:59) |   18 sec (00:00:18) | 2727 sec (00:45:27) |

HPCCSmoketest commented:
Process of PR-13955, label: h24018 starts now.
The reason of this test is: New commit.
Commit ID: 52e79e1993cf142726b23ded0c87a1bbe2a65227
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 52e79e1993cf142726b23ded0c87a1bbe2a65227
Build: success
Milestone:Install hpccsystems-platform-community_7.10.3-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |74 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |28 sec  (00:00:28) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |883 |883 |0 |577 sec  (00:09:37) |
|test (thor) |821 |821 |0 |1041 sec  (00:17:21) |
|test (roxie) |962 |962 |0 |716 sec  (00:11:56) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) | 1055 sec (00:17:35) |  124 sec (00:02:04) |   23 sec (00:00:23) |   17 sec (00:00:17) | 2685 sec (00:44:45) |   17 sec (00:00:17) | 3930 sec (01:05:30) |

wangkx commented:
@afishbeck you may check my changes.

afishbeck commented on esp/services/ws_workunits/ws_workunitsHelpers.hpp:
```diff
@@ -134,11 +134,14 @@ struct WsWUExceptions
 
 class WsWuInfo
 {
+    std::set<unsigned> serviceActivities;
```
Very picky, but the name "serviceActivities" could be kinds or nodes.. maybe call this "serviceActivityKinds"?

afishbeck reviewed (commented):
@wangkx one fairly trivial comment, otherwise looks good.

wangkx commented:
@afishbeck Renamed.

HPCCSmoketest commented:
Process of PR-13955, label: h24018 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 68b29510edcfbfc9c3934a2274f306184048853c
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-230.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-13955, label: h24018 starts now.
The reason of this test is: New commit.
Commit ID: 68b29510edcfbfc9c3934a2274f306184048853c
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-230.ca-central-1.compute.internal
Sha: 68b29510edcfbfc9c3934a2274f306184048853c
Build: success
Milestone:Install hpccsystems-platform-community_7.10.5-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |73 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |43 sec  (00:00:43) |
|setup (roxie) |11 |11 |0 |16 sec  (00:00:16) |
|test (hthor) |883 |883 |0 |313 sec  (00:05:13) |
|test (thor) |821 |821 |0 |748 sec  (00:12:28) |
|test (roxie) |962 |962 |0 |391 sec  (00:06:31) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  745 sec (00:12:25) |  118 sec (00:01:58) |   22 sec (00:00:22) |   17 sec (00:00:17) | 1797 sec (00:29:57) |   19 sec (00:00:19) | 2727 sec (00:45:27) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 68b29510edcfbfc9c3934a2274f306184048853c
Build: success
Milestone:Install hpccsystems-platform-community_7.10.5-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |74 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |883 |883 |0 |578 sec  (00:09:38) |
|test (thor) |821 |821 |0 |1041 sec  (00:17:21) |
|test (roxie) |962 |962 |0 |715 sec  (00:11:55) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) | 1051 sec (00:17:31) |  126 sec (00:02:06) |   23 sec (00:00:23) |   18 sec (00:00:18) | 2708 sec (00:45:08) |   18 sec (00:00:18) | 3953 sec (01:05:53) |

jakesmith commented on esp/services/ws_workunits/ws_workunitsHelpers.hpp:
```diff
@@ -134,10 +134,14 @@ struct WsWUExceptions
 
 class WsWuInfo
 {
+    std::set<unsigned> serviceActivityKinds;
```
@wangkx - an unordered_set is more efficient.

jakesmith commented on esp/services/ws_workunits/ws_workunitsHelpers.cpp:
```diff
@@ -898,6 +899,54 @@ void WsWuInfo::getGraphTimingData(IArrayOf<IConstECLTimingData> &timingData)
+    Owned<IConstWUGraphIterator> graphs = &cw->getGraphs(GraphTypeActivities);
+    ForEach(*graphs)
+    {
+        Owned <IPropertyTree> xgmml = graphs->query().getXGMMLTree(false);
```
trivial: extra space after Owned

jakesmith commented on esp/services/ws_workunits/ws_workunitsHelpers.cpp:
```diff
@@ -898,6 +899,54 @@ void WsWuInfo::getGraphTimingData(IArrayOf<IConstECLTimingData> &timingData)
+                continue;
+
+            IArrayOf<IEspNamedValue> attrs;
+            IPropertyTreeIterator* attIterator = treeNode.getElements("att");
```
leaks

jakesmith reviewed (changes requested):
would be worth fixing the leak.

HPCCSmoketest commented:
Process of PR-13955, label: h24018 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 128e197dd937b2966571257715a0b48c8d72144e
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-25.ca-central-1.compute.internal

wangkx commented:
@jakesmith thanks for the review. Could you please review my fixes?

jakesmith reviewed (approved):
@wangkx - looks good.

jakesmith commented:
Please squash.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-25.ca-central-1.compute.internal
Sha: 128e197dd937b2966571257715a0b48c8d72144e
Build: success
Milestone:Install hpccsystems-platform-community_7.10.5-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |73 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |27 sec  (00:00:27) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |25 sec  (00:00:25) |
|test (hthor) |883 |883 |0 |309 sec  (00:05:09) |
|test (thor) |821 |821 |0 |754 sec  (00:12:34) |
|test (roxie) |962 |962 |0 |409 sec  (00:06:49) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   10 sec (00:00:10) |  746 sec (00:12:26) |  119 sec (00:01:59) |   23 sec (00:00:23) |   18 sec (00:00:18) | 1824 sec (00:30:24) |   18 sec (00:00:18) | 2758 sec (00:45:58) |

wangkx commented:
@jakesmith squashed

HPCCSmoketest commented:
Process of PR-13955, label: h24018 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 2b51c44ad5516b63383fab08f398fc8355070755
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-191.ca-central-1.compute.internal

afishbeck reviewed (approved):
@wangkx looks good

wangkx commented:
@ghalliday this PR is ready to be merged.

HPCCSmoketest commented:
Process of PR-13955, label: h24018 starts now.
The reason of this test is: New commit.
Commit ID: 2b51c44ad5516b63383fab08f398fc8355070755
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-191.ca-central-1.compute.internal
Sha: 2b51c44ad5516b63383fab08f398fc8355070755
Build: success
Milestone:Install hpccsystems-platform-community_7.10.5-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |73 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |31 sec  (00:00:31) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |16 sec  (00:00:16) |
|test (hthor) |883 |883 |0 |329 sec  (00:05:29) |
|test (thor) |821 |821 |0 |798 sec  (00:13:18) |
|test (roxie) |962 |962 |0 |431 sec  (00:07:11) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  743 sec (00:12:23) |  118 sec (00:01:58) |   23 sec (00:00:23) |   21 sec (00:00:21) | 1909 sec (00:31:49) |   19 sec (00:00:19) | 2842 sec (00:47:22) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 2b51c44ad5516b63383fab08f398fc8355070755
Build: success
Milestone:Install hpccsystems-platform-community_7.10.5-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |76 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |30 sec  (00:00:30) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |22 sec  (00:00:22) |
|test (hthor) |883 |883 |0 |581 sec  (00:09:41) |
|test (thor) |821 |821 |0 |1053 sec  (00:17:33) |
|test (roxie) |962 |962 |0 |715 sec  (00:11:55) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) | 1060 sec (00:17:40) |  125 sec (00:02:05) |   23 sec (00:00:23) |   17 sec (00:00:17) | 2729 sec (00:45:29) |   18 sec (00:00:18) | 3980 sec (01:06:20) |

ghalliday commented:
@wangkx how does this fit in with https://track.hpccsystems.com/browse/HPCC-23970?
It looks like it is trying to solve the same problem, and the information should now be available to Gordon through an existing call.

There are a couple of issues with this PR - it doesn't work with child queries, and it is likely the graph format is going to change in the near to medium term, so it would be good to not add new code that relies on the current format.

ghalliday commented:
(If the information is added to this service, it should internally use the same code as wu details to extract the list.)

wangkx commented:
> how does this fit in with https://track.hpccsystems.com/browse/HPCC-23970? It looks like it is trying to solve the same problem, and the information should now be available to Gordon through an existing call.
@GordonSmith does the https://track.hpccsystems.com/browse/HPCC-24018 duplicate https://track.hpccsystems.com/browse/HPCC-23970?

GordonSmith commented:
@wangkx It looks like @ghalliday PR only attaches the information to the WU - your PR is still needed as its exposes the ESPmethod - but I suspect you should possible just read the new information rather than gathering it your self?

ghalliday commented:
@GordonSmith You should be able to extract it using the appropriate wudetails call.  However, a modified version of this PR is needed if you want it returned by Wuinfo.  Which do you want?

GordonSmith commented:
@ghalliday Maybe both - It may be a misconception, but I consider the WUDetails calls to be "heavy", WUInfo calls to be "medium" and WUQuery calls to be "light".  If you happy I can use WUDetails (with appropriate conditions via ESP) for things like this, I should probably start using it more?

ghalliday commented:
Anything that reads the entire graph is not really light, or medium, weight.  I think for the moment @wangkx should implement this using the getScopeIterator() functionality, but we should consider extracting the information so it can be accessed more efficiently (either some kind of cache, or cloning this information).

wangkx commented:
> wangkx should implement this using the getScopeIterator() functionality.

@ghalliday I see that the 'serviceName' att has been added into some graph node. How can I use the to getScopeIterator() to get the 'serviceName' att from graph node? Which WuScopeFilter should I use?

ghalliday commented:
@wangkx it looks like you can't currently filter the scope iterator to only those items with a particular attribute.  (You can filter by statistic.)  

I have opened a jira (HPCC-24423) and will fix it.  For the moment I would use
addOutputAttribute(WaServiceName);
and then call iter->queryAttribute(WaServiceName) - which may currently come back null until I have implemented the filter.

ghalliday commented:
PR released

wangkx committed:
HPCC-24018 Add IncludeServiceNames option to WUInfo

Default to false.

Revise based on review:
1. Remove the case WaServiceName;
2. Change the WuScopeFilter filter;
3. Add/check the return value from the queryAttribute().

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

wangkx commented:
@ghalliday I just created a WU with a SOAPCALL WsWorkunits.WUAction using the code from your PR. I can see the <att name="serviceName" value="WUAction"/>inside graph node, There is no 'serviceName' inside the Statistic section. Is that correct behavior?
Then, I added the following code into my WUInfo. The queryAttribute(WaServiceName) returns null. Is there anything wrong in my code?
WuScopeFilter filter("source[global],depth[1,]");
    filter.sourceFlags = SSFsearchGraphStats;
    filter.addOutputAttribute(WaServiceName);
    Owned<IConstWUScopeIterator> it = &cw->getScopeIterator(filter);
    ForEach(*it)
    {
        StringBuffer s;
        DBGLOG("####aaaaa(%s)(%s)(%s)", it->queryScope(), queryScopeTypeName(it->getScopeType()), it->queryAttribute(WaServiceName, s));
    }

wangkx commented:
Looks like the filter.sourceFlags = SSFsearchAll helps.

HPCCSmoketest commented:
Process of PR-13955, label: h24018 starts now.
The reason of this test is: New commit.
Commit ID: 761474c6d7b9dfbcf751a906a20ade3589517e4e
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-13955, label: h24018 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 761474c6d7b9dfbcf751a906a20ade3589517e4e
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-61.ca-central-1.compute.internal

wangkx commented:
@ghalliday this PR has been changed to use the getScopeIterator() functionality. Please review

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-61.ca-central-1.compute.internal
Sha: 761474c6d7b9dfbcf751a906a20ade3589517e4e
Build: success
Milestone:Install hpccsystems-platform-community_7.10.5-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |72 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |28 sec  (00:00:28) |
|setup (thor) |11 |11 |0 |43 sec  (00:00:43) |
|setup (roxie) |11 |11 |0 |21 sec  (00:00:21) |
|test (hthor) |883 |883 |0 |311 sec  (00:05:11) |
|test (thor) |821 |821 |0 |761 sec  (00:12:41) |
|test (roxie) |962 |962 |0 |408 sec  (00:06:48) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  740 sec (00:12:20) |  118 sec (00:01:58) |   23 sec (00:00:23) |   18 sec (00:00:18) | 1831 sec (00:30:31) |   19 sec (00:00:19) | 2758 sec (00:45:58) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 761474c6d7b9dfbcf751a906a20ade3589517e4e
Build: success
Milestone:Install hpccsystems-platform-community_7.10.5-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |76 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |30 sec  (00:00:30) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |883 |883 |0 |586 sec  (00:09:46) |
|test (thor) |821 |821 |0 |1039 sec  (00:17:19) |
|test (roxie) |962 |962 |0 |716 sec  (00:11:56) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) | 1061 sec (00:17:41) |  124 sec (00:02:04) |   23 sec (00:00:23) |   17 sec (00:00:17) | 2721 sec (00:45:21) |   17 sec (00:00:17) | 3971 sec (01:06:11) |

ghalliday commented on esp/services/ws_workunits/ws_workunitsHelpers.cpp:
```diff
@@ -898,6 +898,28 @@ void WsWuInfo::getGraphTimingData(IArrayOf<IConstECLTimingData> &timingData)
+        return;
+
+    StringArray serviceNames;
+    WuScopeFilter filter("depth[1,]");
```
You don't really need the depth filter - it will only filter a tiny fraction of the items, and will cost more to check.

For the filter you should be able to either use the string, or specify particular attributes.  You also shouldn't need to specify the source if you have provided enough information.

So you should either use
   WuScopeFilter filter("attr[servicename]");
or
   WuScopeFilter filter;
   filter.addOutputAttribute(WaServiceName);
   filter.finishedFilter();

Once the other PR has been merged this should be:
   WuScopeFilter filter("where[servicename],attr[servicename]");
or
   WuScopeFilter filter;
   filter.addOutputAttribute(WaServiceName);
   filter.addRequiredAttribute(WaServiceName);
   filter.finishedFilter();

There should also be a call to filter.finishedFilter();

wangkx replied:
I thought that specifying a source will limit the search area. I will change.

ghalliday commented on esp/services/ws_workunits/ws_workunitsHelpers.cpp:
```diff
@@ -898,6 +898,28 @@ void WsWuInfo::getGraphTimingData(IArrayOf<IConstECLTimingData> &timingData)
+
+    StringArray serviceNames;
+    WuScopeFilter filter("depth[1,]");
+    filter.sourceFlags = SSFsearchGraph;
```
Only the public member functions should be used.  I will investigate why those member variables are not private

wangkx replied:
Ok

ghalliday commented on esp/services/ws_workunits/ws_workunitsHelpers.cpp:
```diff
@@ -898,6 +898,28 @@ void WsWuInfo::getGraphTimingData(IArrayOf<IConstECLTimingData> &timingData)
+            continue;
+
+        StringBuffer serviceName;
+        it->queryAttribute(WaServiceName, serviceName);
```
This needs to be

    const char * value = it->queryAttribute(WaServiceName, serviceName);
    if (value)

There is no guarantee that the parameter passed as a scratchpad will be modfied.

wangkx replied:
Will change

ghalliday commented on common/workunit/wuattr.cpp:
```diff
@@ -247,6 +247,9 @@ extern WORKUNIT_API const char * queryAttributeValue(IPropertyTree & src, WuAttr
     case WaIdRemoteSubGraph:
         value = scratchpad.clear().append(ChildGraphScopePrefix).append(value).str();
         break;
+    case WaServiceName: //Should be default:?
```
This should be deleted - you cannot rely on scratchpad being modified

wangkx replied:
Will delete

ghalliday reviewed (commented):
@w

ghalliday commented:
@wangkx

HPCCSmoketest commented:
Process of PR-13955, label: h24018 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 86e291a038e828da5aea219e2ae00dcd570bcb5f
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-151.ca-central-1.compute.internal

wangkx commented:
@ghalliday please review my changes.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-151.ca-central-1.compute.internal
Sha: 86e291a038e828da5aea219e2ae00dcd570bcb5f
Build: success
Milestone:Install hpccsystems-platform-community_7.10.5-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |73 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |25 sec  (00:00:25) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |17 sec  (00:00:17) |
|test (hthor) |883 |883 |0 |312 sec  (00:05:12) |
|test (thor) |821 |821 |0 |760 sec  (00:12:40) |
|test (roxie) |962 |962 |0 |407 sec  (00:06:47) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  746 sec (00:12:26) |  118 sec (00:01:58) |   23 sec (00:00:23) |   18 sec (00:00:18) | 1823 sec (00:30:23) |   18 sec (00:00:18) | 2755 sec (00:45:55) |

HPCCSmoketest commented:
Process of PR-13955, label: h24018 starts now.
The reason of this test is: New commit.
Commit ID: 86e291a038e828da5aea219e2ae00dcd570bcb5f
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 86e291a038e828da5aea219e2ae00dcd570bcb5f
Build: success
Milestone:Install hpccsystems-platform-community_7.10.5-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |73 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |32 sec  (00:00:32) |
|setup (thor) |11 |11 |0 |46 sec  (00:00:46) |
|setup (roxie) |11 |11 |0 |22 sec  (00:00:22) |
|test (hthor) |883 |883 |0 |590 sec  (00:09:50) |
|test (thor) |821 |821 |0 |1041 sec  (00:17:21) |
|test (roxie) |962 |962 |0 |729 sec  (00:12:09) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) | 1050 sec (00:17:30) |  124 sec (00:02:04) |   23 sec (00:00:23) |   17 sec (00:00:17) | 2719 sec (00:45:19) |   17 sec (00:00:17) | 3959 sec (01:05:59) |

ghalliday commented on esp/services/ws_workunits/ws_workunitsHelpers.cpp:
```diff
@@ -898,6 +898,28 @@ void WsWuInfo::getGraphTimingData(IArrayOf<IConstECLTimingData> &timingData)
+    Owned<IConstWUScopeIterator> it = &cw->getScopeIterator(filter);
+    ForEach(*it)
+    {
+        if (it->getScopeType() != SSTactivity)
```
This filter should be added to the WuScopeFilter, rather than being checked in the iterator loop:

    filter.addScopeType("activity");

I have also merged the PR that allows a restriction on the attribute, so you can also add

    filter.addRequiredAttr(WaServiceName);

ghalliday commented:
@wangkx sorry for the delay - I was sure I had replied.  It looks good - one further comment.

HPCCSmoketest commented:
Process of PR-13955, label: h24018 starts now.
The reason of this test is: New commit.
Commit ID: 12a484f3368926613a2710165e2b57a5e152432f
Estimated completion time is ~0.9 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-13955, label: h24018 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 12a484f3368926613a2710165e2b57a5e152432f
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-16.ca-central-1.compute.internal

wangkx commented:
@ghalliday I just made the changes and rebased the commits to the latest 7.10.x. Please review.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-16.ca-central-1.compute.internal
Sha: 12a484f3368926613a2710165e2b57a5e152432f
Build: success
Milestone:Install hpccsystems-platform-community_7.10.7-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |73 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |26 sec  (00:00:26) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |16 sec  (00:00:16) |
|test (hthor) |883 |883 |0 |310 sec  (00:05:10) |
|test (thor) |821 |821 |0 |771 sec  (00:12:51) |
|test (roxie) |962 |962 |0 |407 sec  (00:06:47) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  751 sec (00:12:31) |  119 sec (00:01:59) |   23 sec (00:00:23) |   18 sec (00:00:18) | 1832 sec (00:30:32) |   20 sec (00:00:20) | 2772 sec (00:46:12) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 12a484f3368926613a2710165e2b57a5e152432f
Build: success
Milestone:Install hpccsystems-platform-community_7.10.7-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |77 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |28 sec  (00:00:28) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |22 sec  (00:00:22) |
|test (hthor) |883 |883 |0 |585 sec  (00:09:45) |
|test (thor) |821 |821 |0 |1045 sec  (00:17:25) |
|test (roxie) |962 |962 |0 |724 sec  (00:12:04) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) | 1050 sec (00:17:30) |  125 sec (00:02:05) |   23 sec (00:00:23) |   18 sec (00:00:18) | 2710 sec (00:45:10) |   17 sec (00:00:17) | 3951 sec (01:05:51) |

ghalliday commented:
@wangkx looks good.  Please squash and I will merge.

wangkx commented:
@ghalliday Squashed. Please merge.

HPCCSmoketest commented:
Process of PR-13955, label: h24018 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 7be4af71b4cc48ced1746cc052ddae3c71f63ef1
Estimated completion time is ~0.8 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-149.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-149.ca-central-1.compute.internal
Sha: 7be4af71b4cc48ced1746cc052ddae3c71f63ef1
Build: success
Milestone:Install hpccsystems-platform-community_7.10.7-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |144 |144 |0 |0 |0 |76 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |25 sec  (00:00:25) |
|setup (thor) |11 |11 |0 |43 sec  (00:00:43) |
|setup (roxie) |11 |11 |0 |25 sec  (00:00:25) |
|test (hthor) |883 |883 |0 |308 sec  (00:05:08) |
|test (thor) |821 |821 |0 |771 sec  (00:12:51) |
|test (roxie) |962 |962 |0 |413 sec  (00:06:53) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  764 sec (00:12:44) |  120 sec (00:02:00) |   23 sec (00:00:23) |   18 sec (00:00:18) | 1846 sec (00:30:46) |   18 sec (00:00:18) | 2798 sec (00:46:38) |
